gen3users/validation.py

```diff
@@ -2,6 +2,21 @@
 import yaml
 
 
+def assert_and_log(assertion_success, error_message):
+    """
+    If an assertion fails, logs the provided error message and updates
+    the global variable "failed_validation" for future use.
+    
+    Args:
+        assertion_success (bool): result of an assertion.
+        error_message (str): message to display if the assertion failed.
+    """
+    global failed_validation
```

Comment:
I'm really averse to `global` (maybe excessively) so I just want to propose an alternative pattern, using closures:
```python
def validate_user_yaml(user_yaml, name="user.yaml"):
    failed = False
    def assert_and_log(success, msg):
        if not success:
            failed = True
            logger.error(msg)
```

Reply:
(also IMO using the actual `logging` would be better than `print`s)

Reply:
I'm not sure if this would lead to too much reorganization in this case, so for this PR a better way might just be something like this:
```python
def validate_everything():
    ok = validate_groups() and validate_policies() and validate_users()
    if not ok:
        raise Exception("...")

def validate_groups():
    ok = True
    ok = ok and assert_and_log(condition, msg)
    ...
    return ok

def assert_and_log(ok, msg):
    if not ok:
        logger.error(msg)
    return ok
```
It's more functional and probably seems a bit tedious but I think this is somewhat clearer

Reply:
if `assert_and_log ` is a nested function, all the `validate_x` functions have to be nested too

i'd like to be able to call any individual validation function by itself later if we want to, so i'll use your 2nd approach